app/controllers/course/registrations_controller.rb

```diff
@@ -0,0 +1,129 @@
+
+    # TODO: Notify lecturer
+    redirect_to course_path(@course), flash: { notice: t('.request_format') %
+      { role: t("course.registrations.role.#{role}") } }
+  end
+
+  def update #:nodoc:
+    if params[:update_single]
+      update_registration
+    elsif params[:approve_single]
+      approve_registration
+    elsif params[:delete_single]
+      redirect_to delete_course_registration_path(@course, @registration)
+    else
+      redirect_to course_path(@course), status: :bad_request
+    end
+  end
+
+  def delete #:nodoc:
+    message = t('.notice_format') % { role: t("course.registrations.role.#{@registration.role}"),
+                                      user: @registration.user.name }
+    @registration.destroy!
+    redirect_to course_registrations_path(@course), flash: { notice: message }
+  end
+
+  def update_multiple #:nodoc:
+    if should_process_single?
+      redirect_to update_course_registration_path(@course,
+                                                  params[:approve_single] || params[:delete_single],
+                                                  params)
+    else
+      process_multiple_registrations
+    end
+
+  end
+
+  private
+
+  def should_process_single? #:nodoc:
+    params[:approve_single] || params[:delete_single]
+  end
+
+  def should_process_all? #:nodoc:
+    params[:approve_all] || params[:delete_all]
+  end
+
+  def should_approve? #:nodoc:
+    params[:approve_single] || params[:approve_selected] || params[:approve_all]
+  end
+
+  def update_scope #:nodoc:
+    params[:approve_all] || params[:delete_all]
+  end
+
+  def update_registration #:nodoc:
+    @registration.update_attributes(registration_params)
+
+    redirect_to course_path(@course),
+                flash: { notice: t('.update_notice_format') %
+                  { role: t("course.registrations.role.#{@registration.role}") } }
+  end
+
+  def approve_registration #:nodoc:
+    flash = {}
+    begin
+      flash[:notice] = t('.approve_notice_format') %
+        { role: t("course.registrations.role.#{@registration.role}"),
+          user: @registration.user.name }
+      Course::Registration.process!(@course, [@registration.id], approve: true)
+    rescue
+      flash[:error] = t('.error_format') % { reason: $ERROR_INFO }
+    end
+
+    redirect_to course_registrations_path(@course), flash: flash
+  end
+
+  def process_multiple_registrations
+    @flash = {}
+    processed_ids = Course::Registration.process!(
+      @course, params[:registration_ids], approve: should_approve?,
+      process_all: should_process_all?, role: update_scope)
+    @flash[:notice] = (should_approve?) ? t('.approve_notice_format') : t('.delete_notice_format')
+    @flash[:notice] %= { count: processed_ids.count }
+  rescue
+    @flash[:error] = t('.error_format') % { reason: $!.message }
+  ensure
+    redirect_to course_registrations_path(@course), flash: @flash
+  end
+
+  def ensure_course_open #:nodoc:
+    unless @course.opened?
```

Comment:
Use a guard clause instead of wrapping the code inside a conditional expression.